src/main/java/edu/harvard/iq/dataverse/SettingsWrapper.java

```diff
@@ -161,35 +196,91 @@ public String getGuidesBaseUrl() {
     }
 
     public String getGuidesVersion() {
-        return systemConfig.getGuidesVersion();
+        if (guidesVersion == null) {
+            String saneDefault = getAppVersion();
+            guidesVersion = getValueForKey(SettingsServiceBean.Key.GuidesVersion, saneDefault);
+        }
+        return guidesVersion;
+    }
+    
+    // OK to call SystemConfig - the method does not rely on database 
+    // settings.
+    public String getDataverseSiteUrl() {
+        if (siteUrl == null) {
+            siteUrl = systemConfig.getDataverseSiteUrl();
+        }
+        return siteUrl;
     }
     
+    // OK to call SystemConfig - the value is based on a JVM option; no 
+    // extra database lookups. 
     public Long getZipDownloadLimit(){
-        return systemConfig.getZipDownloadLimit();
+        if (zipDownloadLimit == null) {
+            String zipLimitOption = getValueForKey(SettingsServiceBean.Key.ZipDownloadLimit);
+            zipDownloadLimit = SystemConfig.getLongLimitFromStringOrDefault(zipLimitOption, SystemConfig.defaultZipDownloadLimit);
+        }
+        return zipDownloadLimit;
     }
 
     public boolean isPublicInstall(){
-        return systemConfig.isPublicInstall();
+        if (publicInstall == null) {
+            boolean saneDefault = false;
+            publicInstall = isTrueForKey(SettingsServiceBean.Key.PublicInstall, saneDefault);
+        }
+        return publicInstall; 
     }
     
     public boolean isRsyncUpload() {
-        return systemConfig.isRsyncUpload();
+        if (rsyncUpload == null) {
+            rsyncUpload = getUploadMethodAvailable(SystemConfig.FileUploadMethods.RSYNC.toString());
+        }
+        return rsyncUpload; 
     }
     
     public boolean isRsyncDownload() {
-        return systemConfig.isRsyncDownload();
+        if (rsyncDownload == null) {
+            String downloadMethods = getValueForKey(SettingsServiceBean.Key.DownloadMethods);
+            rsyncDownload = downloadMethods != null && downloadMethods.toLowerCase().contains(SystemConfig.FileDownloadMethods.RSYNC.toString());
+        }
+        return rsyncDownload;
     }
     
```

Comment:
OK, so my original rationale for duplicating these methods was pure penny-, or rather, query-pinching. Performing a check like this (that relies on looking up a single Setting) in the wrapper - as opposed to calling SystemConfig and caching the result - saves 1 (One) query total. (Because the wrapper already has all the settings in memory, looked up all at once; and SystemConfig will in fact have to look up that setting in the database. 

After way more deliberation than this issue deserved, I am undoing the duplication of logic in this, and a couple of other methods. These few query lookups - from a small, heavily cached table - are not worth the drawbacks of duplicating the code, and potentially ending up with the pages and the API getting different configuration results. 

Saving these queries may be achieved in the future via some application-scoped caching mechanism, as mentioned elsewhere in the PR.